aggregator/aggregator.go

```diff
@@ -197,26 +228,58 @@ func aggregate(context *cli.Context) error {
 				}
 			}
 
-			if len(previous) == 0 || stateChanged {
+			aboveThreshold := (float64(eligibleNodeCount)/float64(totalNodeCount))*100 > float64(thresholdPercentage)
+			toggle = toggle && aboveThreshold
+
+			// This is the check for first aggregation cycle. No previous state exist at this point.
```

Comment:
We should be calculating the count directly from the nomad API, it looks like this PR even includes getEligibleNodeCount for fetching this.

Reply:
The `count` is computed directly from the Nomad API. The `previous` state map is for health checks. It makes sure to toggle/untoggle the node only if there is a `state change` e.g. if `docker` was unhealthy in the previous aggregation cycle, and we took the node out of scheduling pool in the previous aggregation cycle. If, in the current aggregation cycle `docker` is **still** unhealthy, there is no need to call Nomad API again, since there is no state change (`Unhealthy` --> `Unhealthy`)